core/domain/exp_domain.py

```diff
@@ -2499,3 +2499,470 @@ def add_contribution_by_user(self, contributor_id):
+                elif (change.property_name ==
+                      STATE_PROPERTY_CARD_IS_CHECKPOINT):
+                    change_is_mergeable = True
+                elif (change.property_name ==
+                      STATE_PROPERTY_RECORDED_VOICEOVERS):
+                    if state_name in self.changed_properties:
+                        if (all(property not in
+                                self.changed_properties[state_name]
+                                for property in
+                                (self
+                                 .PROPERTIES_CONFLICTING_VOICEOVERS_CHANGES
+                                )) and
+                                STATE_PROPERTY_RECORDED_VOICEOVERS not in
+                                self.changed_properties[state_name]):
+                            change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+            elif change.cmd == CMD_ADD_WRITTEN_TRANSLATION:
+                state_name = change.state_name
+                if change.state_name in state_names_of_renamed_states:
+                    state_name = (
+                        state_names_of_renamed_states[change.state_name])
+                if state_name in old_to_new_state_names:
+                    # Here we will send the changelist, frontend_version,
+                    # backend_version and exploration to the admin, so
+                    # that the changes related to state renames can be
+                    # reviewed and the proper conditions can be written
+                    # to handle those cases.
+                    return False, True
+                if state_name not in self.changed_translations:
+                    self.changed_translations[state_name] = []
+                if state_name in self.changed_properties:
+                    changed_property = self._get_property_name_from_content_id(
+                        change.content_id)
+                    if (changed_property not in
+                            (self.changed_properties[state_name] +
+                             self.changed_translations[state_name])):
+                        change_is_mergeable = True
+                else:
+                    change_is_mergeable = True
+            elif (change.cmd ==
+                  CMD_MARK_WRITTEN_TRANSLATIONS_AS_NEEDING_UPDATE):
+                change_is_mergeable = True
+            elif change.cmd == CMD_EDIT_EXPLORATION_PROPERTY:
+                change_is_mergeable = (
+                    exp_at_change_list_version.__getattribute__(
+                        change.property_name) ==
+                    current_exploration.__getattribute__(
+                        change.property_name))
+
+            if change_is_mergeable:
+                changes_are_mergeable = True
+                continue
+            else:
+                changes_are_mergeable = False
+                break
```

Comment:
This can not be done for now because we use the default reject technique for every change so suppose if we remove change_is_mergeable and then we'll need to use continue so if we use continue and if the last change is not mergeable then we'll exit the loop and how we'll now whether the last changes was mergeable or not? I'll create an issue for this and we'll improve this in the future versions of this functionality.

Reply:
#13373